triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerImpl.scala

```diff
@@ -80,6 +82,10 @@ object TriggerRunnerImpl {
           case Status(replyTo) =>
             replyTo ! QueryingACS
             Behaviors.same
+          case QueryACSFailed(cause: io.grpc.StatusRuntimeException)
+              if cause.getStatus == io.grpc.Status.UNAUTHENTICATED =>
+            config.server ! Server.TriggerTokenExpired(triggerInstance)
```

Comment:
This feels a bit weird. We are bypassing the supervisor here and sending it directly to the server. Have you considered throwing an exception instead and handling the child terminating in the supervisor instead?

Reply:
Looking more deeply into this unveiled a [bug in akka-actor-typed](https://github.com/akka/akka/issues/29841) where the restart supervisor triggers even if an underlying stop supervisor should stop the actor. This means we cannot throw an exception from `TriggerRunnerImpl` to generate a corresponding `ChildFailed` signal on `TriggerRunner`. I've now worked around this (and a related issue on `InitializationHalted`) with a custom `BehaviorInterceptor` that then sends a message to the `TriggerRunner`.

The `TriggerRunner` in turn could throw an exception and let the server handle failure due to token expiry in a `ChildFailed` signal handler. However, this is awkward to implement as we need to forward additional information to the server so that it can properly restart the trigger. Namely `UUID`, `Trigger`, and `CompiledPackages`. We could store these in the exception, but that seems too heavy for an exception. Instead, the `TriggerRunner` sends a message to the server requesting token refresh and restart and then shuts itself down.